src/LSST/control/ControlMain.cpp

```diff
@@ -57,16 +66,161 @@ ControlMain::Ptr ControlMain::getPtr() {
     return _thisPtr;
 }
 
-ControlMain& ControlMain::get() {
-    if (_thisPtr == nullptr) {
-        throw system::ConfigException(ERR_LOC, "ControlMain has not been setup.");
+ControlMain::ControlMain() {}
+
+ControlMain::~ControlMain() {}
+
+void ControlMain::run(int argc, const char* argv[]) {
+        thread t(&ControlMain::_cMain, this, argc, argv);
+        _mainThrd = move(t);
+}
+
+
+void signalHandler(int sig) {
+    if (sig == SIGPIPE) {
+        LWARN("Ignoring SIGPIPE sig=", sig);
+        return;
     }
-    return *_thisPtr;
+    LCRITICAL("Unhandled signal caught sig=", sig);
+    exit(sig);
 }
 
-ControlMain::ControlMain() {}
+void ControlMain::_cMain(int argc, const char* argv[]) {
+    LINFO("starting main");
+    util::Log& log = util::Log::getLog();
 
-ControlMain::~ControlMain() {}
+    // Get the configuration
+    system::Config& sysCfg = system::Config::get();
+
+    // Setup global items.
+    system::Globals::setup(sysCfg);
+
+    // Setup a simple signal handler to handle when clients closing connection results in SIGPIPE.
+    signal(SIGPIPE, signalHandler);
+
+    // Create the control system
+    _simCore.reset(new LSST::m2cellcpp::simulator::SimCore());
+    _simCore->start();
+    faultmgr::FaultMgr::setup();
+    control::FpgaIo::setup(_simCore);
+    control::MotionEngine::setup();
+    control::Context::setup();
+
+    // Register the PowerSystem with FpgaIo so that it gets updates.
+    auto powerSys = control::Context::get()->model.getPowerSystem();
+    control::FpgaIo::get().registerPowerSys(powerSys);
+
+    // Setup the control system
+    auto context = control::Context::get();
+    context->model.ctrlSetup();
+
+    // At this point, MotionCtrl and FpgaCtrl should be configured. Start the control loops
+    context->model.ctrlStart();
+
+    // At this point, the LabView code seems to want to put the system into
+    // ReadyIdle. We're NOT doing that. The system is going into StandbyState
+    // until it gets an explicit command to do something else.
+
+    // Start the telemetry server
+    // FUTURE: Telemetry for the GUI requires the ComControlServer->ComConnection::welcomeMsg to
+    //         work properly. Those elements should be in the telemetry so there's no need for a
+    //         ComControlServer connection for Telemetry. This has to wait until the existing
+    //         controller is no longer used.
+    LINFO("Starting Telemetry Server");
+    // FUTURE: get the correct entries into `system::Config`.
+    int const telemPort = 50001;
+    auto servTelemetryMap = system::TelemetryMap::Ptr(new system::TelemetryMap());
+    auto telemetryServ = system::TelemetryCom::create(servTelemetryMap, telemPort);
+
+    telemetryServ->startServer();
```

Comment:
Looks like a redundancy in the naming convention: "server starts server". The following should be enough:
```
telemetryServ->start();
```

Reply:
The redundancy is intentional. When working with control systems, the available tools are often not very good, so unique names can help locate things when limited to tools like grep.